app/services/solidus_bolt/base.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module SolidusBolt
+  class Base
+    def initialize(*)
+      @config = SolidusBolt::BoltConfiguration.fetch
+    end
+
+    def call
+      raise NotImplementedError
+    end
+
+    def self.call(*args)
+      new(*args).call
+    end
+
+    private
+
+    def api_base_url
+      @config.environment_url
```

Comment:
so in the end we aren't using `attr_reader`?

Reply:
We'll use it in the other classes that I have removed for now from the PR